pkg/controller/registry/reconciler/grpc.go

```diff
@@ -327,3 +335,48 @@ func (c *GrpcRegistryReconciler) CheckRegistryServer(catalogSource *v1alpha1.Cat
 	healthy = true
 	return
 }
+
+// promoteCatalog swaps the labels on the update pod so that the update pod is now reachable by the catalog service.
+// By updating the catalog on cluster it promotes the update pod to act as the new version of the catalog on-cluster.
+func (c *GrpcRegistryReconciler) promoteCatalog(updatePod *corev1.Pod, key string) error {
+	// Update the update pod to promote it to serving pod via the SSA client
```

Comment:
should this drop the label from the old pod before adding to the new? there might be a small period of time where the service is routing to both pods

Reply:
this is a good point - in our existing implementation we promote the update pod and then delete the old serving pod. so there is a period of overlap, and this PR does not change that design. I've thought about the issue myself back when traveling to our F2F. 

right now its
```
update labels on update pod -> (both pods are serving) -> delete serving pod -> only update pod left serving
```
we could have
```
remove labels on serving pod -> (no pod is serving) -> update labels on update pod -> only update pod serving -> delete serving pod
```

The latter is probably a bit cleaner, especially if we can apply both label changes via SSA so they occur basically simultaneously. 

However, from the point of view of the catalog client (the subscription essentially), it doesn't really know when the catalog has been updated, and so a small period where it gets either the new or the old content doesn't really affect it. There's no clear way for the user to know "ok, now my catalog has been updated, i can make a new sub". They just have a rough view of it based on their polling interval. Since the feature is more meant to provide a means for updating existing subs the new operator will get onto the cluster eventually. 

tl;dr i think this could be optimized but I'm not sure it improves much based on the current state of the feature